bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoDiscoveryService.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    @Override
+    public void startScan() {
+        BindingConfiguration localConf = config;
+        ApiBridgeHandler localHandler = handler;
+        if (localHandler != null && localConf != null) {
+            ThingUID apiBridgeUID = localHandler.getThing().getUID();
+            try {
+                AircareApi airCareApi = localHandler.getRestManager(AircareApi.class);
+                if (airCareApi != null) { // Search Healthy Home Coaches
+                    ListBodyResponse<NAMain> body = airCareApi.getHomeCoachData(null).getBody();
+                    if (body != null) {
+                        body.getElements().stream().forEach(homeCoach -> createThing(homeCoach, apiBridgeUID));
+                    }
+                }
+                if (localConf.readFriends) {
+                    WeatherApi weatherApi = localHandler.getRestManager(WeatherApi.class);
+                    if (weatherApi != null) { // Search favorite stations
+                        ListBodyResponse<NAMain> body = weatherApi.getStationsData(null, true).getBody();
+                        if (body != null) {
+                            body.getElements().stream().filter(NAMain::isReadOnly).forEach(station -> {
+                                ThingUID bridgeUID = createThing(station, apiBridgeUID);
+                                station.getModules().values().stream()
+                                        .forEach(module -> createThing(module, bridgeUID));
+                            });
+                        }
+                    }
+                }
+                HomeApi homeApi = localHandler.getRestManager(HomeApi.class);
+                if (homeApi != null) { // Search all the rest
+                    homeApi.getHomesData(null, null).stream().filter(h -> !h.getFeatures().isEmpty()).forEach(home -> {
+                        ThingUID homeUID = createThing(home, apiBridgeUID);
+                        home.getKnownPersons().forEach(person -> createThing(person, homeUID));
+                        home.getModules().values().stream().forEach(device -> {
+                            ModuleType deviceType = device.getType();
+                            String deviceBridge = device.getBridge();
+                            ThingUID bridgeUID = deviceBridge != null && deviceType.getBridge() != ModuleType.HOME
+                                    ? findThingUID(deviceType.getBridge(), deviceBridge, apiBridgeUID)
+                                    : deviceType.getBridge() == ModuleType.HOME ? homeUID : apiBridgeUID;
+                            createThing(device, bridgeUID);
+                        });
+                        home.getRooms().values().stream().forEach(room -> {
+                            room.getModuleIds().stream().map(id -> home.getModules().get(id))
+                                    .map(m -> m != null ? m.getType().feature : FeatureArea.NONE)
```

Comment:
Can m be null ?

Reply:
In principle...no, but the compiler complains of a potential NPE.